### ISSUE TITLE
Upgrade to latest Electron

### ISSUE THREAD

**Issue by twolfson** [twolfson/google-music-electron#44]
It's been a while since we last upgraded Electron so let's get to it. We started to do this in:

https://github.com/twolfson/google-music-electron/issues/39#issuecomment-273439733

but ran into an issue. This issue to keep tabs on the reason/progress

**Comment by twolfson** [comment]
We have identified the problem to being Electron@1.2.0 by running vanilla Electron with Google Music until we stopped getting failures

```bash
npm install electron-prebuilt@1.3.x --save
./node_modules/.bin/electron
# window.location = 'http://music.google.com/';

npm install electron-prebuilt@1.2.x --save
npm install electron-prebuilt@1.1.x --save
```

After reading the changelog for the release, the `beforeunload` bindings stuck out to us:

https://github.com/electron/electron/releases/tag/v1.2.0

We tried out removing the `beforeunload` bindings from Google Music and sure enough it started closing properly:

```js
// 4 listeners
window.getEventListeners(window).beforeunload

// Remove listeners and find out it works
// DEV: We ran the following 4 times
window.getEventListeners(window).beforeunload[0].remove()
```

**Comment by twolfson** [comment]
It's actually the 3rd listener that seems to cause the issue by they are all cryptic =(

We are going to upgrade to latest Electron to verify it's the same problem

**Comment by twolfson** [comment]
Yep, same problem on `1.4.x`. Glad we caught this issue, now to read the docs...

**Comment by twolfson** [comment]
We have a poor fix via `gme.browserWindow.on('close')` and `process.exit` but it's not ideal as `closed` would be nicer/more accurate =/

**Comment by twolfson** [comment]
I'm too tired to trust my current judgement but I'm going to open a thread on Electron to get their feedback

**Comment by twolfson** [comment]
Ah, damn. We need to ask on Slack and I definitely don't have time/consciousness for that =(

**Comment by jacobped** [comment]
Get some sleep. You can take a look at it later when you're up for it.

**Comment by twolfson** [comment]
Heh, thanks @jacobped. I didn't mention it in the comments but I was planning on logging off once those tasks were completed =P

**Comment by twolfson** [comment]
Note to self: Clicking on task icon broke for restoring minimized window in latest Electron =/

**Comment by twolfson** [comment]
Started looking into this again. It looks like pausing music allows closing to work again as well. This means we can likely fix the issue with a hack that pauses music on quit request

I also wonder if we can stop `beforeunload` propagation somehow...

**Comment by twolfson** [comment]
Oooh, and we have a string in the event: `e.returnValue = 'You still have music playing.';`

```js
window.onbeforeunload = function (e) { console.log(e); }

/*
BeforeUnloadEvent {isTrusted: true, returnValue: "You still have music playing.", type: "beforeunload", target: document, currentTarget: Window…}
bubbles: false
cancelBubble: false
cancelable: true
composed: false
currentTarget: Window
defaultPrevented: false
eventPhase: 2
isTrusted: true
path: Array[1]
returnValue: "You still have music playing."
srcElement: document
target: document
timeStamp: 49002.850000000006
type: "beforeunload"
__proto__: BeforeUnloadEvent
*/
```

**Comment by twolfson** [comment]
Unfortunately, attempting to edit the `returnValue` and use `stopPropagation` is in vain =(

```js
window.onbeforeunload = function (e) { e.returnValue = false; console.log(e); }
window.onbeforeunload = function (e) { e.returnValue = true; console.log(e); }
window.onbeforeunload = function (e) { delete e.returnValue; console.log(e); }
window.onbeforeunload = function (e) { e.stopPropagation(); delete e.returnValue; console.log(e); }
window.onbeforeunload = function (e) { e.returnValue = ''; console.log(e); }
```

**Comment by twolfson** [comment]
Posted to Slack, now we twiddle our thumbs until there's a response :tada: 

https://atomio.slack.com/archives/electron/p1484801505004057

**Comment by twolfson** [comment]
We haven't made any headway on this. We have decided to proxy `window.addEventListener` and call it a day. This has been resolved/released in `2.15.0`
